packages/azure-http-specs/specs/azure/resource-manager/operation-templates/optional-body.tsp

```diff
@@ -0,0 +1,152 @@
+  @doc("The name to update.")
+  name?: string;
+
+  @doc("The description to update.")
+  description?: string;
+}
+
+model ActionRequest {
+  @doc("The action type to perform.")
+  actionType?: string;
+
+  @doc("Additional action parameters.")
+  parameters?: string;
+}
+
+model ActionResult {
+  @doc("The result of the action.")
+  result: string;
+}
+
+@armResourceOperations
+interface OptionalBody {
+  @scenario
+  @scenarioDoc("""
+    Resource GET operation to retrieve a widget.
+    
+    Expected verb: GET
+    Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.OperationTemplates/widgets/widget1
+    Expected query parameter: api-version=2023-12-01-preview
+    Expected status code: 200
+    Expected response body:
+    ```json
+    {
+      "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.OperationTemplates/widgets/widget1",
+      "name": "widget1",
+      "type": "Azure.ResourceManager.OperationTemplates/widgets",
+      "location": "eastus",
+      "properties": {
+        "name": "widget1",
+        "description": "A test widget",
+        "provisioningState": "Succeeded"
+      },
+      "systemData": {
+        "createdBy": "AzureSDK",
+        "createdByType": "User",
+        "createdAt": <any date>,
+        "lastModifiedBy": "AzureSDK",
+        "lastModifiedAt": <any date>,
+        "lastModifiedByType": "User"
+      }
+    }
+    ```
+    """)
+  get is ArmResourceRead<Widget>;
+
+  @scenario
+  @scenarioDoc("""
+    Resource PATCH operation using Legacy.CustomPatchSync with no request body.
+    This tests the optional body functionality where the request body is not sent.
+    
+    Expected verb: PATCH  
+    Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.OperationTemplates/widgets/widget1
+    Expected query parameter: api-version=2023-12-01-preview
+    Expected request body: None (empty body)
+    Expected status code: 200
+    Expected response body:
+    ```json
+    {
+      "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.OperationTemplates/widgets/widget1",
+      "name": "widget1",
+      "type": "Azure.ResourceManager.OperationTemplates/widgets",
+      "location": "eastus",
+      "properties": {
+        "name": "widget1",
+        "description": "A test widget",
+        "provisioningState": "Succeeded"
+      },
+      "systemData": {
+        "createdBy": "AzureSDK",
+        "createdByType": "User",
+        "createdAt": <any date>,
+        "lastModifiedBy": "AzureSDK",
+        "lastModifiedAt": <any date>,
+        "lastModifiedByType": "User"
+      }
+    }
+    ```
+    """)
+  updateWithoutBody is Azure.ResourceManager.Legacy.CustomPatchSync<
+    Widget,
+    UpdateRequest,
+    OptionalRequestBody = true
+  >;
+
```

Comment:
I meant to say ArmProviderActionAsync/ArmProviderActionSync.

Reply:
Seems there's another issue tracking this?
https://github.com/Azure/typespec-azure/issues/2835